viewing_party/party.py

```diff
@@ -0,0 +1,184 @@
+# Wave 1
+
+def create_movie(movie_title, genre, rating):  
+  new_movie = {}
+  
+  if not movie_title or not genre or not rating:
+    return None  
+  
+  else:
+    new_movie["title"] = movie_title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+    
+  return new_movie
+
+
+def add_to_watched(user_data, movie):
+
+  if not "watched" in user_data:
+    user_data["watched"] = [movie]
+    
+  else: 
+    user_data["watched"].append(movie)
+    
+  return user_data
+  
+  
+def add_to_watchlist(user_data, movie):
+  updated_dict = user_data
+  
+  if not "watchlist" in user_data:
+    user_data["watchlist"] = [movie]
+    
+  else: 
+    user_data["watchlist"].append(movie)
+    
+  return user_data
+
+
+def watch_movie(user_data, title):
+  
+  movie_found = False
+  index_to_be_removed = -1  # set to -1 bcs invalid, can't set it to other bcs those would be valid
+  
+  for index, temp_movie_dict in enumerate(user_data["watchlist"]):
+    if temp_movie_dict["title"] == title: #found movie to be removed
+      index_to_be_removed = index #the index of the movie
+      movie_found = True
+      user_data["watched"].append(temp_movie_dict)
```

Comment:
Since this code implies that it's only going to find one movie (index_to_be_removed can only be a single value), you could `break` out of this loop as soon as you find the first thing to remove. We don't _expect_ there to be more than one copy of a movie with that title, but you never know...

BTW, excellent to see an approach like this which avoids modifying the collection being iterated over during the iteration!